internal/filters/main.go

```diff
@@ -0,0 +1,101 @@
+/*
+A CLI->Go template shim used to render arbitrary sprig functions.
+
+To Run:
+> go run main.go sprigFunctionName sprigFunctionArguments...
+
+Example:
+> go run main.go trim "      some string      "
+
+This program takes textual input and attempts to resolve the sprigFunctionArguments types through inspecting
+sprigFunctionName's signature.  Bad input results in fatal exit with error code 1.  Debug output is intentionally
+suppressed as stdout is utilized by the calling program.
+*/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Masterminds/sprig"
+	"os"
+	"reflect"
+	"strconv"
+)
+
+const fatalExitReturnCode = 1
+const funcArgsCLIArgsFirstIndex = 2
+const funcNameCLIArgsIndex = 1
+const funcReturnOutputIndex = 0
+
+// Wrapper to call a Golang function through reflection.
+func CallFunc(funcMap map[string]interface{}, name string, arguments ... interface{}) (result []reflect.Value,
+	err error) {
+
+	function := reflect.ValueOf(funcMap[name])
+	if len(arguments) != function.Type().NumIn() {
+		err = errors.New("number of arguments is not adapted")
+		return
+	}
+	in := make([]reflect.Value, len(arguments))
+	for k, param := range arguments {
+		in[k] = reflect.ValueOf(param)
+	}
+	result = function.Call(in)
+	return
+}
+
+// Exits 1 with an appropriate message.
+func fatalError(message string) {
+	fmt.Fprintln(os.Stderr, "ERROR: " + message)
+	os.Exit(fatalExitReturnCode)
+}
+
+// Ensures that a sprig/template function was provided as a CLI argument.
+func checkForFuncName(args *[]string) {
+	if len(*args) < 2 {
+		fatalError("No Sprig Function name was provided")
+	}
+}
+
+// Invokes the CLI shim.  The first argument is the sprig/template function name.  The remaining arguments are the
+// sprig/template function arguments.
+func main() {
+	args := os.Args
+	// This would indicate a function with no funcName argument.
+	checkForFuncName(&args)
+
+	funcName := args[funcNameCLIArgsIndex]
+	funcArgs := args[funcArgsCLIArgsFirstIndex:]
+
+	// Loads only the text/template and Sprig Function Map.
+	funcMap := sprig.TxtFuncMap()
+
+	// Used to convert CLI args into functional arguments.
+	var iFuncArgs []interface{}
+	f := reflect.ValueOf(funcMap[funcName])
+	var argType = reflect.String
+	for i, b := range funcArgs {
+		// Attempt to derive the argument type from the method signature, and cast/convert the input appropriately.
+		if i < f.Type().NumIn() {
```

Comment:
I think this is where you should handle panic

Reply:
I resolved the panic by checking that the type is in fact a Function in a subsequent patch, per my comment above.  Additionally, I added error checking for failed casting.

This conditional is actually still needed;  this logic is attempting to cast the input arguments from **string** to match whatever is specified by the method signature.  However, some methods take variable length arguments, so there is certainly a point at which the CLI string argument index will exceed whatever is provided by **f.Type().In(i)**.  In this case, we know that whatever the argument type was uncovered during the last valid call to **f.Type.In(i)** will be the argument type for the rest of the var args.